Add GitHub issue templates for bugs, features, support and PRs

dragetd opened the pull request:
Issues are either a bug report, feature request or support request. Labels
are automaticaly added to the issues respectively: bug, enhancement, support.

The templates suggest using the Slack before opening an issue, but no longer
say it is a hard requirement. This might give a more inclusive impression to
people reporting who are not Slack users.

- [x] a conversation was held in the appropriate [SDKMAN Slack](https://slack.sdkman.io) channel.
- [x] a Github Issue was opened for this feature / bug.
- [ ] test coverage was added (Cucumber or Spock as appropriate).

This addresses Issue #753

CONTRIBUTING.md was also rephrased.

The tag 'support' would be new. We could not set a tag there. Or even rephrase the text to be more 'We do not want to offer support via GitHub AT ALL', if requested.

I have set it to the default branch of my fork, so you can test how it looks via this repository: https://github.com/draget-forks/sdkman-cli (feel free to create test-issues or PRs in there!)

marc0der commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,18 @@
+<!-- Steps to reproduce the behavior or verify the issue -->
+
+**System info**
+<!-- Please add relevant information about your system, e.g. Win/Linux, OS version, (WSL?), etc. Please also include the output of `sdk version` -->
```
Also important is the shell (zsh/bash) and its version.

marc0der commented on .github/ISSUE_TEMPLATE/support_request.md:
As above, the shell info will also be useful here.

dragetd commented:
I rebased and updated the branch. The new template can be checked here: https://github.com/draget-forks/sdkman-cli/issues/new/choose

I'd say it is worth a try. If it creates more confusion than it helps, it can be easily reverted. Thank you for the constructive discussion!

marc0der commented:
Thanks for contributing these!
